src/Storage/Entity/Content.php

```diff
@@ -57,7 +57,9 @@ public function set($key, $value)
         $setter = 'set' . ucfirst($key);
         if (is_array($value)) {
             // Filter empty values from the array
-            $value = array_filter($value);
+            $value = array_filter($value, function ($item) {
+                return ($item !== null) && ($item !== '');
+            });
```

Comment:
This doesn't do anything. See the [PHP docs](http://php.net/manual/en/function.array-filter.php):

> If no callback is supplied, all entries of array equal to FALSE (see [converting to boolean](http://php.net/manual/en/language.types.boolean.php#language.types.boolean.casting)) will be removed.

Easy enough to see:
```
$value = [null, false, ''];

$value = array_filter($value);
dump($value); 
// Outputs []

$value = array_filter($value, function ($item) {
    return ($item !== null) && ($item !== '');
});
dump($value);
// Outputs []
```

Reply:
Actually, as i told in issue,  i was fixing `[..., 0, ...]` case. I didn't know exactly, what was the purpose of filtering at all - so i fixed my actual case.
@GawainLynch can you take a look at investigation in #7411 ?

Reply:
Derp! Yeah, that makes sense.

Although what I would say is that I don't think the bug as reported is valid due to a number of reasons. But I haven't had time to look yet :innocent:

Reply:
Ookay, so please take your time) and notify me on update)

Reply:
Maybe.. you could just merge this PR )?

Reply:
@Boorj We should never, ever "just merge" a PR, if we have an inkling that we don't fully grasp the consequences. If we'd do that, there's much potential for opening ourselves up to other inexplainable bugs down the road. 

This might 'resolve' some of the wrong result, but it doesn't fix the _cause_. :-)